### ISSUE TITLE
LAMMPS interface

### ISSUE THREAD

**Issue by shyuep** [materialsvirtuallab/matgl#111]
An interface to LAMMPS for MatGL is needed. This can be done in two ways:

1. Using LAMMPS's experimental python plugin infrastructure. See https://github.com/lanl/hippynn/blob/development/hippynn/interfaces/lammps_interface/mliap_interface.py for an example. This would not be very efficient code but it would be the easiest to finish quickly and allow users to leverage on LAMMPS workflows.
2. Write a C interface to LAMMPS. This was done for the TF version of M3GNet and we can perhaps adapt that implementation.

**Comment by jan-janssen** [comment]
What about the LAMMPS interface by advancesoftcrop https://github.com/advancesoftcorp/lammps/tree/based-on-lammps_2Jun2022/src/ML-M3GNET ?  As far as I can see it is compatible to matgl but internally it still calls python, so the performance is presumably similar to the performance of the ASE interface.

**Comment by SmallBearC** [comment]
I am also looking forward an interface to LAMMPS. Especially, I need to train multiple models and use the model-devision feature to better sample. I have no better way for instance.

**Comment by shyuep** [comment]
Yes, the AdvancedSoft interface works. But we are hoping to develop a more universal solution for MatGL itself.

**Comment by duanshaodong123** [comment]
Hello, this web page can't open https://github.com/advancesoftcorp/lammps/tree/based-on-lammps_2Jun2022/src/ML-M3GNET. Can you provide other downloading methods

> What about the LAMMPS interface by advancesoftcrop https://github.com/advancesoftcorp/lammps/tree/based-on-lammps_2Jun2022/src/ML-M3GNET ? As far as I can see it is compatible to matgl but internally it still calls python, so the performance is presumably similar to the performance of the ASE interface.

**Comment by bawanglong-2** [comment]
bai5@bai5:~/testlammps/FeH$ mpirun -np 16 lmp_mpi -in  run.in
LAMMPS (2 Aug 2023)
ERROR: Pair style M3GNet does not support MPI parallelization (../pair_m3gnet.cpp:221)
Last command: pair_style    m3gnet /home/bai5/software/lammps-based-on-lammps_2Aug2023/potentials/M3GNET/
bai5@bai5:~/testlammps/FeH$
as it

**Comment by advancesoftcorp** [comment]
I have changed the package of LAMMPS from ML-M3GNET to ML-GNNP, which has other GNNPs.

https://github.com/advancesoftcorp/lammps/tree/based-on-lammps_2Aug2023/src/ML-GNNP
